pkgs/development/compilers/go/1.13.nix

```diff
@@ -156,16 +156,18 @@ stdenv.mkDerivation rec {
   GOHOSTOS = stdenv.buildPlatform.parsed.kernel.name;
   GOHOSTARCH = goarch stdenv.buildPlatform;
 
-  # {CC,CXX}_FOR_TARGET must be only set for cross compilation case as go expect those
+  # {CC,CXX}_FOR_TARGET must be set for cross compilation as go expect those
   # to be different from CC/CXX
+  # We explicitely set it to the default case because we set CC further down.
+  # The default values can be found in src/cmd/dist/build.go in the go source.
   CC_FOR_TARGET = if (stdenv.buildPlatform != stdenv.targetPlatform) then
       "${pkgsTargetTarget.stdenv.cc}/bin/${pkgsTargetTarget.stdenv.cc.targetPrefix}cc"
     else
-      null;
+      if stdenv.isDarwin then "clang" else "gcc";
```

Comment:
```suggestion
      if stdenv.cc.isClang then "clang" else "gcc";
```

Reply:
just `cc` is available everywhere, making the conditional unnecessary

Reply:
I'm going off the Go default found here: https://github.com/golang/go/blob/9f33108dfa22946622a8a78b5cd3f64cd3e455dd/src/cmd/dist/build.go#L206-L209